numpy/lib/npyio.py

```diff
@@ -561,6 +561,11 @@ def savez(file, *args, **kwds):
     its list of arrays (with the ``.files`` attribute), and for the arrays
     themselves.
 
+    When saving dictionaries, avoid using keys that begin with "/", as the
+    first slash will be removed by the python archive function. If they 
+    must be used, prepend another character ahead of the slash to ensure
+    that the "/" will not be removed.
```

Comment:
So if variable names contain `..`, you're in for a bad time too. Surprisingly, it seems that the reserved windows file name (like `nul`) aren't actually an issue here.